### PR TITLE
various: cleanup of "inherit version;"

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
a bit related to #103997

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130196` at 70238dd2 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built successfully:</summary>
  <ul>
    <li>direnv</li>
    <li>edid-generator</li>
    <li>grml-zsh-config</li>
    <li>passExtensions.pass-tomb</li>
    <li>tomb</li>
    <li>zsh</li>
    <li>zsh-autosuggestions</li>
    <li>zsh-powerlevel10k</li>
    <li>zsh-syntax-highlighting</li>
    <li>zthrottle</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/edid-generator/default.nix:29:3:
```
   |
29 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/tools/misc/edid-generator/default.nix:27:3:
```
   |
27 |   configurePhase = (lib.concatMapStringsSep "\n" (m: "echo \"${m}\" | ./modeline2edid -") modelines);
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/shells/zsh/default.nix [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
   pname = "zsh";
-  inherit version;
+  version = "5.8";
 
   src = fetchurl {
     url = "mirror://sourceforge/zsh/zsh-${version}.tar.xz";
     sha256 = "09yyaadq738zlrnlh1hd3ycj1mv3q5hh4xl1ank70mjnqm6bbi6w";
   };
 
+  documentation = fetchurl {
```
Comment (roberth) [inline pkgs/shells/zsh/default.nix:12]: If you're not using the `$documentation` bash variable, you shouldn't create it.
Please revert.

File: pkgs/tools/graphics/twilight/default.nix [hunk 2]
```diff
@@ -1,11 +1,9 @@
 { lib, stdenv, fetchFromGitHub
 , libGL, libGLU, freeglut, libX11 }:
 
-let
-  version = "2018-04-19";
-in stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "twilight";
-  inherit version;
+  version = "2018-04-19";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/twilight/default.nix:6]:
```suggestion
  version = "unstable-2018-04-19";
```

File: pkgs/tools/misc/hdaps-gl/default.nix [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-let version = "0.0.7"; in
-stdenv.mkDerivation {
-      pname = "hdaps-gl";
-      inherit version;
-      src = fetchFromGitHub {
-            owner = "linux-thinkpad";
-            repo = "hdaps-gl";
-            rev = version;
-            sha256 = "0jywsrcr1wzkjig5cvz014c3r026sbwscbkv7zh1014lkjm0kyyh";
-      };
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hdaps-gl/default.nix:3]:
```suggestion
```

File: pkgs/tools/misc/edid-decode/default.nix [hunk 4]
```diff
@@ -1,9 +1,8 @@
 { lib, stdenv, fetchgit }:
-let
-  version = "2018-12-06";
-in stdenv.mkDerivation {
+
+stdenv.mkDerivation rec {
   pname = "edid-decode-unstable";
-  inherit version;
+  version = "2018-12-06";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/edid-decode/default.nix:5]:
```suggestion
  version = "unstable-2018-12-06";
```

File: pkgs/shells/zsh/default.nix (hunk 1 above)
Comment (Stunkymonkey) [reply pkgs/shells/zsh/default.nix:12]: it is used in line 35

Comment (roberth) [reply pkgs/shells/zsh/default.nix:12]: Line 35 uses the Nix variable, so it doesn't have to be added to bash via `mkDerivation { documentation = x; }`. Please put it back in its `let` binding.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 130196` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>aliceml</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>direnv</li>
    <li>edid-decode</li>
    <li>edid-generator</li>
    <li>gitstatus</li>
    <li>grml-zsh-config</li>
    <li>passExtensions.pass-tomb</li>
    <li>timedoctor</li>
    <li>tomb</li>
    <li>twilight</li>
    <li>vimPlugins.direnv-vim</li>
    <li>zsh</li>
    <li>zsh-autosuggestions</li>
    <li>zsh-powerlevel10k</li>
    <li>zsh-syntax-highlighting</li>
    <li>zthrottle</li>
  </ul>
</details>

**Pull request merged**